### ISSUE TITLE
Does not work well with jest 27

### ISSUE THREAD

**Issue by Naddiseo** [mswjs/interceptors#123]
[Jest 27 removes `setImmediate`](https://github.com/facebook/jest/blob/master/CHANGELOG.md#fixes-2), which is used in [createClientRequestOverride](https://github.com/mswjs/interceptors/blob/f95c5c2d6028e22bd0681695bc4887c89cac81bc/src/interceptors/ClientRequest/createClientRequestOverride.ts#L124). 

Currently, I get the following in my tests:
```
ReferenceError: setImmediate is not defined
          at ClientRequestOverride.write (~/node_modules/@mswjs/interceptors/src/interceptors/ClientRequest/createClientRequestOverride.ts:124:7)
          at writeToStream (~/node_modules/node-fetch/lib/index.js:653:8)
          at ~/node_modules/node-fetch/lib/index.js:1627:3

```

**Comment by jonkoops** [comment]
More information can be found in https://github.com/facebook/jest/issues/11204 and https://github.com/facebook/jest/pull/11222.

**Comment by kettanaito** [comment]
Thanks for reporting this, @Naddiseo. The fix has been opened in #124, I'll try to review it and release it this week.
